woodwork/datatable.py

```diff
@@ -423,7 +424,11 @@ def drop(self, columns):
         if not_present:
             raise ValueError(f'{not_present} not found in DataTable')
 
-        return self._new_dt_from_cols([col for col in self._dataframe.columns if col not in columns])
+        dt = self._new_dt_from_cols([col for col in self._dataframe.columns if col not in columns])
+        if any('index' in self[column].semantic_tags for column in columns):
```

Comment:
Can we use the index property here to simplify this code?
`if self.index in columns:`